### PR TITLE
Add Goals & Roadmap to README

### PR DESCRIPTION
**Pull request by moshloop**
/assign @timothysc @dims @detiber @figo @codenrhoden @randomvariable @dlipovetsky @neolit123 @justinsb

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,3 +14,43 @@ You can reach the maintainers of this project at:
 ### Code of conduct
 
 Participation in the Kubernetes community is governed by the [Kubernetes Code of Conduct](code-of-conduct.md).
+
+
+
+## Goals 
+* To build images for Kubernetes-conformant clusters in a consistent way across infrastructures, providers, and business needs.
+   * Install all software, containers, and configuration needed to pass conformance tests.
```
Comment (neolit123) [inline README.md]: shouldn't passing conformance be the responsibility of CAPI?

File: README.md [hunk 2]
```diff
@@ -14,3 +14,43 @@ You can reach the maintainers of this project at:
...
+   * Support end users requirements to customize images for their business needs.
+* To provide assurances in the binaries and configuration in images for purposes of security auditing and operational stability.
+   * Allow introspection of artifacts, software versions, and configurations in a given image.
+   * Support repeatable build processes where the same inputs of requested install versions result in the same installed binaries.
+* To ensure that the creation of images is performed via well defined phases.  Where users could choose specific phases that they needed.
+    * Support incremental usage.
```
Comment (neolit123) [inline README.md]: given there is only one item in the list, include it as part of the the above sentence?

File: README.md [hunk 3]
```diff
@@ -14,3 +14,43 @@ You can reach the maintainers of this project at:
...
+
+## Non-Goals 
+* To publish images to cloud provider marketplaces, or to provide software workflow to automatically upload the built images on the cloud provider infrastructure.
+    * For example, it is not the responsibility of *this* utility to publish images to Amazon Marketplace. Each Cluster API Provider may implement its own image publisher. Users should be able to use the provider's publisher with the image output by the image builder.
+* To provide upgrade or downgrade semantics.
+* To provide guarantees that the software installed provides a fully functional system.
+* To prescribe the hardware architecture of the build system.
+* To create images from scratch.  
+    * The purpose of the tool is to take an existing image and make it Kubernetes ready.
```
Comment (neolit123) [inline README.md]: possibly combine with the previous point.

File: README.md [hunk 4]
```diff
@@ -14,3 +14,43 @@ You can reach the maintainers of this project at:
 ### Code of conduct
 
 Participation in the Kubernetes community is governed by the [Kubernetes Code of Conduct](code-of-conduct.md).
+
+
+
+## Goals
```
Comment (neolit123) [inline README.md]: > Cross provider Kubernetes image building utility.

i think the doc needs a minor clarification that this is a Virtual Machine image building utility?

Comment (neolit123) [reply README.md]: possibly the doc should also include some of the meeting details and agenda doc link?

File: README.md (hunk 1 above)
Comment (timothysc) [reply README.md]: You could rephrase the statement to be:  `To install all software, containers, and configuration needed by downstream tools such as Cluster API providers, to enable them to pass conformance tests`

File: README.md (hunk 2 above)
Comment (timothysc) [reply README.md]: We could omit the last one.

File: README.md [hunk 5]
```diff
@@ -14,3 +14,43 @@ You can reach the maintainers of this project at:
...
+* To publish images to cloud provider marketplaces, or to provide software workflow to automatically upload the built images on the cloud provider infrastructure.
+    * For example, it is not the responsibility of *this* utility to publish images to Amazon Marketplace. Each Cluster API Provider may implement its own image publisher. Users should be able to use the provider's publisher with the image output by the image builder.
+* To provide upgrade or downgrade semantics.
+* To provide guarantees that the software installed provides a fully functional system.
+* To prescribe the hardware architecture of the build system.
+* To create images from scratch.  
+    * The purpose of the tool is to take an existing image and make it Kubernetes ready.
+    
+    
+## Roadmap
```
Comment (timothysc) [inline README.md]: Should we prioritize the list below?

Comment (moshloop) [reply README.md]: How we would go about doing that?

Comment (timothysc) [reply README.md]: On the next call talk about roadmap and work as a group to outline priorities.  Priorities in OSS is all about who is willing to sign up todo the work.

File: README.md [hunk 6]
```diff
@@ -14,3 +14,43 @@ You can reach the maintainers of this project at:
...
+
+
+## Goals 
+* To build images for Kubernetes-conformant clusters in a consistent way across infrastructures, providers, and business needs.
+   * Install all software, containers, and configuration needed to pass conformance tests.
+   * Support end users requirements to customize images for their business needs.
+* To provide assurances in the binaries and configuration in images for purposes of security auditing and operational stability.
+   * Allow introspection of artifacts, software versions, and configurations in a given image.
+   * Support repeatable build processes where the same inputs of requested install versions result in the same installed binaries.
+* To ensure that the creation of images is performed via well defined phases.  Where users could choose specific phases that they needed.
```
Comment (dlipovetsky) [inline README.md]: Excellent wordsmithing. Thanks!

**Comment by randomvariable** [comment]
Happy with this. Think the opener of the doc needs to say it's about operating system images as @neolit123 has commented as consumers will confuse it with docker images, but otherwise /lgtm

File: README.md (hunk 3 above)
Comment (codenrhoden) [reply README.md]: Does describing something as a non-goal preclude it? The OVA image creator in `images/capi` does exactly this -- create images from scratch (from an install ISO).

Comment (moshloop) [reply README.md]: Removed as a non-goal as it is the status quo

**Pull request merged**